fourInRowJava.java

```diff
@@ -0,0 +1,126 @@
+                                        {" ", " ", " ", " ", " ", " ", " "}, 
+                                        {" ", " ", " ", " ", " ", " ", " "}, 
+                                        {" ", " ", " ", " ", " ", " ", " "}};
+        
+        public static void printBoard(board b)
+        {
+            for (int col = 0; col < COL; col++)
+            {
+                for (int row = 0; row < ROW; row++)
+                {
+                    System.out.println(TheBoard[col][row] + "");    
+                }
+            }
+        }
+    }
+    
+    class player
+    {
+        static int Row;
+        static int Col;
+    }
+    public class cheakForWin
+    {
+        public static boolean CheakForWin(board b)
+        {
+            for (int row = 0; row < ROW; row++)
+            {
+                if (TheBoard[col][row] == "+" &&
+                    TheBoard[col ][row + 1] == "+" &&
+                    TheBoard[col][row + 2] == "+" &&
+                    TheBoard[col][row + 3] == "+" ||
+                    TheBoard[col][row] == "+" &&
+                    TheBoard[col + 1][row] == "+" &&
+                    TheBoard[col+ 2][row] == "+" &&
+                    TheBoard[col + 3][row] == "+" ||
+                    TheBoard[col][row] == "+" &&
+                    TheBoard[col + 1][row + 1] == "+" &&
+                    TheBoard[col+ 2][row + 2] == "+" &&
+                    TheBoard[col + 3][row + 3] == "+" ||
+                    TheBoard[col][row] == "+" &&
+                    TheBoard[col - 1][row + 1] == "+" &&
+                    TheBoard[col - 2][row + 2] == "+" &&
+                    TheBoard[col - 3][row + 3] == "+")
+                    {
+                        System.out.println("The first player win!!!"); 
+                    }
+                if (TheBoard[col][row] == "-" &&
+                    TheBoard[col ][row + 1] == "-" &&
+                    TheBoard[col][row + 2] == "-" &&
+                    TheBoard[col][row + 3] == "-" ||
+                    TheBoard[col][row] == "-" &&
+                    TheBoard[col + 1][row] == "-" &&
+                    TheBoard[col+ 2][row] == "-" &&
+                    TheBoard[col + 3][row] == "-" ||
+                    TheBoard[col][row] == "-" &&
+                    TheBoard[col + 1][row + 1] == "-" &&
+                    TheBoard[col+ 2][row + 2] == "-" &&
+                    TheBoard[col + 3][row + 3] == "-" ||
+                    TheBoard[col][row] == "-" &&
+                    TheBoard[col - 1][row + 1] == "-" &&
+                    TheBoard[col - 2][row + 2] == "-" &&
+                    TheBoard[col - 3][row + 3] == "-")
+                    {
+                        System.out.println("The second player win!!!"); 
+
+                    }
+            }
+        }
+    }
+    public static void main(String[] args) 
+    {
+        board b = new board();
+        Game g = new Game();
+        player p1 = new player();
```

Comment:
here you use `player` to represent the players but in the first and second functions it is used as a sort of collection of all the moves of the player, a thing that the actual class dose not support.
I think you need to use a data structure inside of the class like a list of some sort to save the move history of the player.

I would also suggest to have some system in place to prevent the players from placing tills underneath existing tills.
in edition to having some player input.